infrastructure/lib/addons/perfdash.ts

```diff
@@ -0,0 +1,241 @@
+import { Construct } from 'constructs';
+import { Aws, CfnStack, StackProps } from 'aws-cdk-lib';
+import { aws_eks as eks } from 'aws-cdk-lib';
+import { aws_iam as iam } from 'aws-cdk-lib';
+
+export interface PerfDashProps extends StackProps {
+  cluster: eks.Cluster;
+  namespace: string;
+}
+
+export class PerfDash extends Construct {
+  constructor(scope: Construct, id: string, props: PerfDashProps) {
+    super(scope, id);
+
+    const ns = props.cluster.addManifest('perfdash-namespace', {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+          name: props.namespace
+      }
+  })
+
+    const sa = props.cluster.addServiceAccount('perfdash-sa', {
+      name: 'perfdash-log-fetcher',
+      namespace: props.namespace
+    });
+
+    sa.node.addDependency(ns)
+    sa.role.attachInlinePolicy(new iam.Policy(this, 'perfdash-policy', {
+      statements: [
+        new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              // S3 readonly access
+              "s3:Get*",
+              "s3:List*",
+              "s3-object-lambda:Get*",
+              "s3-object-lambda:List*",
+            ],
+        }),
+      ],
+  }));
+
+  const cm = props.cluster.addManifest('perfdash-cm', {
+    apiVersion: 'v1',
+    kind: 'ConfigMap',
+    metadata: {
+        name: "perfdash-config",
+        namespace: props.namespace,
+        labels: {
+            "app.kubernetes.io/name": "perfdash",
+            "app.kubernetes.io/component": "config",
+            "app.kubernetes.io/part-of": "perfdash",
+        },
+    },
+    data: {
+        "PERFDASH_LOG_BUCKET": "my-bucket-name",
+        "AWS_DEFAULT_REGION": "us-west-2",
+        "AWS_REGION": "us-west-2",
+        "AWS_ROLE_ARN": sa.role.roleArn,
+        "jobs.yaml": `periodics:
+- name: "kit-eks-1k"
```

Comment:
I feel like config is something that  varies for different customers, I'm not understanding the rationale to cdkfy this over just applying as yaml file through flux from the repo (or) worst case just leave it as is internally like we do today ?

Reply:
Everyone user may need different config in the configmap. I guess it makes sense to remove the configmap out of CDK while AWS_ROLE_ARN can be merged into the Deployment.